docs-v2/reference/sdks/node.mdx

```diff
@@ -459,6 +459,50 @@ await nango.setMetadata('<INTEGRATION-ID>', 'CONNECTION-ID', { 'CUSTOM_KEY1': 'C
 **Response**
 
 Empty response.
+<Expandable>
+```json
+{
+    "CUSTOM_KEY1": "CUSTOM_VALUE1"
+}
+```
+</Expandable>
+
+### Bulk set connection metadata
+
+Set custom metadata for multiple connections (overrides existing metadata).
+
+```js
+await nango.setMetadata('<INTEGRATION-ID>', ['CONNECTION-ID-1', 'CONNECTION-ID-2'], { 'CUSTOM_KEY1': 'CUSTOM_VALUE1' });
+```
+
+**Parameters**
+
+<Expandable>
+    <ResponseField name="providerConfigKey" type="string" required>
+        The integration ID of the connection.
+    </ResponseField>
+
+    <ResponseField name="connectionId" type="string[]" required>
+        The connection IDs.
+    </ResponseField>
+
+    <ResponseField name="metadata" type="Record<string, any>" required>
+        The custom metadata to store in the connection.
+    </ResponseField>
+</Expandable>
+
+**Response**
+
+<Expandable>
+```json
+{
+    "connection_ids": "<string[]>",
+    "metadata": {
+        "CUSTOM_KEY1": "CUSTOM_VALUE1"
+    }
+}
```

Comment:
What if I want to bulk but with different metadata?
I think we could do something like this:
```sh
curl -XPOST /connection/metadata
{ 
  metadatas: [ 
    { connectionId: string, metadata: Record<string, any> },
    { connectionId: string, metadata: Record<string, any> } 
  ]
}
```

Reply:
Seems out of scope of the issue.